os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +255,14 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		mpu_configure_app_regs(&tcb->mpu_regs[STACKGUARD_MPU_NUM * REG_MAX], STACKGUARD_MPU_NUM,
+			(uint32_t)tcb->stack_alloc_ptr + MPU_REGION_SIZE_ALIGNMENT, MPU_REGION_SIZE_ALIGNMENT, true, false);
```

Comment:
Here, the third arguement (region base address) must be tcb->stack_alloc_ptr only. (not tcb->stack_alloc_ptr + MPU_REGION_SIZE_ALIGNMENT)

Reply:
I have removed adding   MPU_REGION_SIZE_ALIGNMENT to the stack_alloc_ptr, as it is lower memory address. MPU region will set 32 bytes from this lower memory towards top of stack.